packages/notebook/src/model.ts

```diff
@@ -191,9 +191,13 @@ export class NotebookModel extends DocumentModel implements INotebookModel {
    */
   toJSON(): nbformat.INotebookContent {
     const cells: nbformat.ICell[] = [];
-    for (let i = 0; i < (this.cells?.length || 0); i++) {
-      const cell = this.cells.get(i);
-      cells.push(cell.toJSON());
+    for (let i = 0; i < (this.cells?.length ?? 0); i++) {
+      const cell = this.cells.get(i).toJSON();
+      if (this._nbformat === 4 && this._nbformatMinor <= 4) {
+        // strip cell ids if we have notebook format 4.0-4.4
+        delete cell.id;
```

Comment:
Tests:
- [ ] if format is 4.4, cell ids are not saved
- [ ] if format is 4.5, cell ids are saved